components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.probes/src/main/java/org/wso2/micro/integrator/probes/ReadinessResource.java

```diff
@@ -96,6 +100,19 @@ public boolean invoke(MessageContext synCtx) {
         }
 
         axisCtx.setProperty(HTTP_SC, CACHED_RESPONSE_CODE);
+        if (HOT_DEPLOYMENT == null) {
+            Parameter hotDeployment = axisCtx.getConfigurationContext().getAxisConfiguration().getParameter(
+                    TAG_HOT_DEPLOYMENT);
+            if (hotDeployment != null) {
+                HOT_DEPLOYMENT = JavaUtils.isTrue(hotDeployment.getValue(), false);
+            }
```

Comment:
Shall we have a  method ( is hotDeploymentEnabled ) to read and have this value in AxisConfiguration and call it in other places.